### PR TITLE
HTTP 204 & 205 should skip the converter.

### PR DESCRIPTION
**Pull request by JakeWharton**
@swankjesse @mattprecious

### CODE REVIEW

File: retrofit/src/main/java/retrofit/RestAdapter.java [hunk 1]
```diff
@@ -290,11 +300,6 @@ private Object parseResult(MethodInfo methodInfo, Response response)
         return response;
       }
 
-      ResponseBody body = response.body();
-      if (body == null) {
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit/RestAdapter.java]: This is never true of an OkHttp response.

File: retrofit/src/main/java/retrofit/RestAdapter.java [hunk 2]
```diff
@@ -291,7 +291,12 @@ private Object parseResult(MethodInfo methodInfo, Response response)
       }
 
       ResponseBody body = response.body();
-      if (body == null) {
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit/RestAdapter.java]: This is never true of an OkHttp response.

File: retrofit/src/main/java/retrofit/RestAdapter.java [hunk 3]
```diff
@@ -291,7 +291,12 @@ private Object parseResult(MethodInfo methodInfo, Response response)
...
+      if (statusCode == 204 || statusCode == 205) {
+        // HTTP 204 No Content "...response MUST NOT include a message-body"
+        // HTTP 205 Reset Content "...response MUST NOT include an entity"
+        if (body.contentLength() > 0) {
+          throw new IllegalStateException(statusCode + " response must not include body.");
```
Comment (swankjesse) [inline retrofit/src/main/java/retrofit/RestAdapter.java]: Not a protocol exception? Somewhat harsh to crash the client when the server made the mistake.

Comment (swankjesse) [inline retrofit/src/main/java/retrofit/RestAdapter.java]: (not sure what's default in Retrofit)

File: retrofit/src/test/java/retrofit/RestAdapterTest.java [hunk 4]
```diff
@@ -71,6 +79,56 @@
...
+    Response response = example.direct();
+    assertThat(response.code()).isEqualTo(204);
+  }
+
+  @Test public void http204WithBodyThrows() {
+    server.enqueue(new MockResponse().setStatus("HTTP/1.1 204 Nothin").setBody("Hey"));
+    try {
+      example.something();
+      fail();
+    } catch (RetrofitError e) {
```
Comment (swankjesse) [inline retrofit/src/test/java/retrofit/RestAdapterTest.java]: Ahhh . . . it gets wrapped. Cool.

**Pull request merged**